File: simulation/switching_strategy/base/base_switching_strategy.py
from abc import ABC, abstractmethod

from simulation.direction import Direction

from ...config import Config
from ...crossing_state import CrossingState


class BaseSwitchingStrategy(ABC):
    def __init__(self):
        super().__init__()

        self.config = Config()

    @abstractmethod
    def calculate_winning_direction(self, crossing_state: CrossingState) -> Direction:
        """Returns direction that should be enabled (green) ASAP."""
        ...

File: simulation/src/crossingLogic/switchingStrategy/base/baseSwitchingStrategy.ts
import Config from '../../config';
import CrossingState from '../../crossingState';
import Direction from '../../direction';

export default abstract class BaseSwitchingStrategy {
	config: Config;
	constructor() {
		this.config = Config.instance;
	}
	/**
	 * Returns direction that should be enabled (green) ASAP.
	 */
	abstract calculateWinningDirection(crossingState: CrossingState): Direction;
}
